medusa/show_updater.py

```diff
@@ -151,7 +151,7 @@ def __del__(self):
 
 class ShowUpdate(db.DBConnection):
```

Comment:
This entire class seems pointless.  

First of all if you're only getting the last update time of an indexer, making it a class called ShowUpdate seems counter-intuitive as it appears like it's the update of that particular show.  

Secondly a class where just a simple function would do seems unnecessary. 

Since this is an attribute related to an indexer, it seems to me it would make more sense as part of the Indexer code, and as such it would make more sense to make it a property.  
Then you could get the last update time from the indexer_api directly, and if it's None then get it from the database.  That way the only time you need to hit the DB for getting the last update time, is when you first get the property from the indexer.  As currently written that would mean that every time the indexer is recreated (which happens a lot) you would have to hit the database, but as its a property, its lazy and would only do so if you needed that data.  Also if we went to a singleton indexer object for each indexer (which we should) then it should cut that down to only a single get per indexer until the next restart.